compiler-js/src/main/java/com/redhat/ceylon/compiler/js/TypeGenerator.java

```diff
@@ -295,8 +295,8 @@ static void interfaceDefinition(final Tree.InterfaceDefinition that, final Gener
         //Don't even bother with nodes that have errors
         if (errVisitor.hasErrors(that))return;
         final Interface d = that.getDeclarationModel();
-        //If it's inside a dynamic interface, don't generate anything
-        if (d.isClassOrInterfaceMember() && ((ClassOrInterface)d.getContainer()).isDynamic())return;
+        //If it's inside a dynamic interface (or for some other reason dynamic), don't generate anything
+        if (d.isDynamic())return;
```

Comment:
This is complete rubbish, sorry. Fixed in 1d1a4c846318914cf897e62943531cbd2e21225a.